### PR TITLE
Jenkins parallel tests

### PR DESCRIPTION
**Pull request by dham**
This makes Jenkins on Linux work. Merging this PR will cause the Jenkins tests to run properly for all branches.

### CODE REVIEW

**Comment by wence-** [comment]
Can we add the coverage checking?  See https://docs.codecov.io/docs/supported-ci-providers

File: Jenkinsfile [hunk 1]
```diff
@@ -43,10 +61,26 @@ cd firedrake/src/firedrake; py.test --cov firedrake --short -v ${TEST_FILES}
...
+    }
+    stage('Codecov'){
+      steps {
+        dir('tmp') {
+          timestamps {
+            sh '''
 . ./firedrake/bin/activate
-cd firedrake/src/dolfin-adjoint; py.test -v tests_firedrake
-"""
+pip install flake8
```
Comment (wence-) [inline Jenkinsfile]: No need to install flake8 here?

Comment (dham) [inline Jenkinsfile]: Oh. Duh. copy paste fail.

**Comment by wence-** [comment]
It's unclear to me where the mac build went (are we still waiting for the new mac mini?).  Additionally, I would expect the merge commit to be tested here (it was previously), but now it looks like the github status is only spewing back a single commit status?

**Comment by dham** [comment]
Note that the PR is not getting built by Jenkins due to some overzealous Jenkins security on branches which change the Jenkinsfile.

**Comment by wence-** [comment]
Ah, makes sense.

**Pull request merged**